landlab/utils/tests/test_stream_length.py

```diff
@@ -0,0 +1,156 @@
+import numpy as np
+import math
+from nose.tools import assert_equal, assert_true, assert_false, assert_raises
+from numpy.testing import assert_array_equal
+
+from landlab import RasterModelGrid, FieldError, HexModelGrid, CLOSED_BOUNDARY
+from landlab.components import FlowAccumulator, FastscapeEroder, FlowDirectorSteepest
+from landlab.utils.stream_length import calculate_stream_length
+
+
+def test_no_flow_recievers():
+    """Test that correct error is raised when no flow recievers are on the grid."""
+    # instantiate a model grid, do not run flow accumulation on it
+    mg = RasterModelGrid(30, 70)
+    # test that the stream length utility will fail because of a ValueError
+    assert_raises(FieldError, calculate_stream_length, mg)
+
+
+def test_no_upstream_array():
+    """Test that correct error is raised when no flow__upstream_node_order."""
+    # instantiate a model grid, do not run flow accumulation on it
+    mg = RasterModelGrid(30, 70)
+    z = mg.add_ones('node','topographic__elevation')
+    fd = FlowDirectorSteepest(mg)
+    fd.run_one_step()
+    # test that the stream length utility will fail because of a ValueError
+    assert_raises(FieldError, calculate_stream_length, mg)
+
+
+def test_stream_length_regular_grid_d8():
+    """Test to demonstrate that stream_length utility works as expected with regular grids"""
+    # instantiate a model grid
+    mg = RasterModelGrid((5, 4), spacing=(1, 1))
+    # instantiate an elevation array
+    z = np.array ([[0,0,0,0],[0,21,10,0],[0,31,20,0],[0,32,30,0],[0,0,0,0]],dtype='float64')
```

Comment:
PEP8 also indicates that a comma should be followed by a space. Also, no need for a space after a function name (e.g. np.array):
```python
z = np.array([[0, 0, 0, 0], [0, 21, 10, 0]...
```
Please adjust other instances incorrect spacing.